### PR TITLE
Asciidocs and refdocs tooling

### PR DESCRIPTION
**Pull request by balopat**
This is in context of #629. 

See generated example: 
 * HTML: https://storage.googleapis.com/balint-skaffold/docs/index.html
 * PDF: https://storage.googleapis.com/balint-skaffold/docs/index.pdf

Documentation tooling requirements:

- Written in a format that is renderable on Github 
- Content is accessible through an outline (sidebar preferably)
- Enables easy PR workflows
- Easy to review changes 
- Easy for a contributor to propose changes 

Additional goals: 

- Composable pages to reduce duplication
- Ability to style/brand 
- Satisfy the following layout goals: 

Layout goals: 

- Have a great, searchable single page HTML with a clear outline (the same for the PDF)
- Support Github as a renderer 
  - Have a full Github landing page (just like currently) 
  - The examples would still live in their separate folders as standalone documented examples, with a good Github specific README with table of contents 
- Keep duplication minimal
  - The first chapter of the reference docs would come mostly from the Github landing page 
  - Examples would be included in the reference docs

### CODE REVIEW

File: docs/index.adoc [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+:icons: font
+:imagesdir: img
+:source-highlighter: pygments
+:pygments-css: style
+:stylesheet: css/skaffold.css
+:linkcss: css/skaffold.css
+:doctype: book
+:github-repo-tree: http://github.com/GoogleContainerTools/skaffold/tree/{commit}
+
+=====
```
Comment (r2d4) [inline docs/index.adoc]: Hide these from the github preview if you can

**Comment by dgageot** [comment]
Sorry @balopat, you have to rebase

File: deploy/cloudbuild-release.yaml [hunk 2]
```diff
@@ -1,13 +1,18 @@
+# using default substitutions, provided by Google Container Builder
+# see: https://cloud.google.com/container-builder/docs/configuring-builds/substitute-variable-values#using_default_substitutions
 steps:
 # Build the container that will do the go build.
   - name: 'gcr.io/cloud-builders/docker'
     args: ['build', '-t', 'gcr.io/k8s-skaffold/skaffold:$TAG_NAME', '-f', 'deploy/skaffold/Dockerfile', '.']
 # Do the go build.
   - name: 'gcr.io/k8s-skaffold/skaffold:$TAG_NAME'
-    args: ['make', 'cross']
+    args: ['make', 'cross', 'docs']
```
Comment (r2d4) [inline deploy/cloudbuild-release.yaml]: this fails @balopat 
docker is not in the integration test image. I would move this to a separate step (I don't want to add asciidoctor to the integration/published skaffold image

Comment (balopat) [reply deploy/cloudbuild-release.yaml]: sounds good. I'll fix it up.

**Pull request merged**